pkg/bitnet/internal/model/tokenizer.go

```diff
@@ -0,0 +1,139 @@
+package model
+
+import (
+	"encoding/json"
+	"errors"
+	"io/fs"
+	"strings"
+)
+
+var (
+	ErrTokenizerNotFound = errors.New("tokenizer file not found")
+	ErrVocabNotLoaded    = errors.New("vocabulary not loaded")
+	ErrUnknownToken      = errors.New("unknown token")
+	ErrUnknownTokenID    = errors.New("unknown token ID")
+)
+
+// Tokenizer handles loading and using the BitNet tokenizer.
+type Tokenizer struct {
+	fs            fs.FS
+	modelPath     string
+	Vocab         map[string]int    `json:"vocab"`
+	Merges        map[string]string `json:"merges"`
+	SpecialTokens map[string]int    `json:"special_tokens"`
+}
+
+// NewTokenizer creates a new Tokenizer instance.
+func NewTokenizer(filesystem fs.FS, modelPath string) (*Tokenizer, error) {
+	if filesystem == nil {
+		return nil, errors.New("filesystem cannot be nil")
```

Comment:
These errors should be created in the `var ( ... )` block above. Point is to not have memory allocations during execution.

Reply:
Still not fixed.